pkg/controller/resourcequota/resource_quota_controller.go

```diff
@@ -416,15 +438,25 @@ func (rq *ResourceQuotaController) replenishQuota(groupResource schema.GroupReso
 	}
 }
 
-// Sync periodically resyncs the controller when new resources are observed from discovery.
-func (rq *ResourceQuotaController) Sync(discoveryFunc NamespacedResourcesFunc, period time.Duration, stopCh <-chan struct{}) {
+// Sync periodically resyncs the controller when new resources are
+// observed from discovery. When new resources are detected, Sync will stop all
+// RQ workers, reset rq.restMapper, and resync the monitors.
+//
+// Note that discoveryClient should NOT be shared with rq.restMapper, otherwise
+// the mapper's underlying discovery client will be unnecessarily reset during
+// the course of detecting new resources.
+func (rq *ResourceQuotaController) Sync(discoveryClient discovery.ServerResourcesInterface, period time.Duration, stopCh <-chan struct{}) {
 	// Something has changed, so track the new state and perform a sync.
 	oldResources := make(map[schema.GroupVersionResource]struct{})
 	wait.Until(func() {
 		// Get the current resource list from discovery.
-		newResources, err := GetQuotableResources(discoveryFunc)
-		if err != nil {
-			utilruntime.HandleError(err)
+		newResources := GetQuotableResources(discoveryClient)
```

Comment:
Yeah, it seems like you really want to flush your discovery cache and it doesn't have to do with the restmapper